site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,293 @@
+
+## Why should I choose Carvel
+
+Kubernetes configuration takes many forms – plain YAML configurations, Helm charts, ytt templates, jsonnet templates, etc. Software running on Kubernetes lives in many different places: a Git repository, an archive over HTTP, a Helm repository, etc.
+
+Kapp-controller, a Carvel tool, provides software authors flexibility to choose their own configuration tools, while providing software consumers with consistent declarative APIs to customize, install, and update software on Kubernetes from various sources.
+
+## Prerequisites
+
+Basic knowledge of imgpkg, kbld, kapp-controller, kapp
+
+[`imgpkg`](https://carvel.dev/imgpkg/): A tool to package, distribute, and relocate your Kubernetes configuration and dependent OCI images as one OCI artifact: a bundle.
+
+[`kbld`](https://carvel.dev/kbld/): kbld incorporates image building and image pushing into your development and deployment workflows.
+
+[`kapp-controller`](https://carvel.dev/kapp-controller/): kapp-controller provides declarative APIs to create, customize, install, and update your Kubernetes applications into packages.
+
+[`kapp`](https://carvel.dev/kapp/): kapp is a cli used to deploy and view groups of Kubernetes resources as “application”.
+
+## Installing Carvel Tools
+
+We'll be using Carvel tools throughout this tutorial, so first we'll install them using `install.sh` script.
+
+```bash
+$ wget -O- https://carvel.dev/install.sh > install.sh
+
+# Inspect install.sh before running...
+$ sudo bash install.sh
+# Install kapp-controller
+$ kubectl apply -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml
+```
+
+--------------------------
+
+## Authoring a Carvel Package
+
+To create a package, we need to create two CRs. We will go through step by step to author an nginx helm chart:
+
+**1. Create PackageMetadata CR**: Package Metadata contains very high level information and description about the package. Multiple versions of a package share same package metadata.
+
+Save the below PackageMetadata CR to a file named `pkgMetadata.yaml`
+```yaml
+
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: PackageMetadata
+metadata:
+  # This will be the name of our package metadata
+  name: nginx.bitnami.vmware.com
+spec:
+  displayName: "Bitnami Nginx Carvel Package"
+  longDescription: "Proxifying Server"
+  shortDescription: "Proxifying Server"
+  categories:
+  - proxy-server
+  providerName: VMWare
+  maintainers:
+  - name: "Carvel"
+  - name: "CarvelInd"
+  - name: "Rohit Aggarwal"
+```
+
+**2. Package Helm Chart**: Package is a combination of configuration metadata and OCI images that informs the package manager what software it holds and how to install itself onto a Kubernetes cluster.
```

Comment:
nit:  open this sentence like "The Package CR is ..."  ,  i think that's a better subject noun-phrase. 

then it's not that the Package CR is a combination of config and OCI image,  it's that it holds a reference to the OCI image, right?  so maybe something like

"The Package CR is config including metadata and references to OCI artifacts that informs the package manager..."

Reply:
Adopted

Reply:
Adopted